rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -452,10 +452,25 @@ pub(crate) enum ControllerRequest {
 enum Status {
     /// Firezone is disconnected.
     Disconnected,
-    /// Firezone is signing in and raising the tunnel.
-    Connecting { start_instant: Instant },
+    /// At least one connection request has failed, due to failing to reach the Portal, and we are waiting for a network change before we try again
+    RetryingConnection {
+        /// The token to log in to the Portal, for retrying the connection request.
+        token: SecretString,
+    },
     /// Firezone is ready to use.
     TunnelReady { resources: Vec<ResourceDescription> },
+    /// Firezone is signing in to the Portal.
+    WaitingForPortal {
```

Comment:
What do we gain from splitting `Connecting` into the two `Waiting` states? `connlib` will internally reconnect to the portal if it fails whilst we have the tunnel.

Currently, the fact that we resolve the IPs on startup is somewhat of an implementation detail and that may change again. For example, if we end up doing something like https://github.com/firezone/firezone/pull/6427, we may resolve them dynamically on startup and thus, creating phoenix-channel will become infallible again.

I would suggest to not let this implementation detail escape the lower layers and just treat the entire time as `Connecting`.

Reply:
It doesn't affect the behavior, but it's nice to see "Connecting to Portal" in the menu if we're trying to debug something

Reply:
If it is just for debugging, could we also achieve the same with logs? My recommendation would be to not treat the initial boot-up of connlib as two phases. Where it failed in its boot-process shouldn't really matter to the clients.

The above state machine kind of suggests that we would be also showing this when we are internally re-connecting to the portal but we don't. That is completely transparent to the client (until we do https://github.com/firezone/firezone/issues/5557).